exporter/logicmonitorexporter/logs_exporter.go

```diff
@@ -0,0 +1,142 @@
+	client, err := e.config.HTTPClientSettings.ToClient(host, e.settings)
+	if err != nil {
+		e.logger.Error("Failed to construct HTTP client", zap.Error(err))
+		return err
+	}
+	authParams := utils.AuthParams{
+		AccessID:    e.config.APIToken.AccessID,
+		AccessKey:   e.config.APIToken.AccessKey,
+		BearerToken: e.config.Headers["Authorization"],
+	}
+	options := []logs.Option{
+		logs.WithLogBatchingDisabled(),
+		logs.WithAuthentication(authParams),
+		logs.WithHTTPClient(client),
+		logs.WithEndpoint(e.config.Endpoint),
+	}
+
+	lli, err := logs.NewLMLogIngest(context.Background(), options...)
+	if err != nil {
+		e.logger.Debug("error in initializing LMLogIngest", zap.Error(err))
+		return err
+	}
+	e.logIngestClient = lli
+	return nil
+}
+
+func (e *logExporter) PushLogData(ctx context.Context, lg plog.Logs) (er error) {
+	e.logger.Debug("Pushing log data to Logicmonitor")
+	resourceLogs := lg.ResourceLogs()
+	for i := 0; i < resourceLogs.Len(); i++ {
+		resourceLog := resourceLogs.At(i)
+		libraryLogs := resourceLog.ScopeLogs()
+		for j := 0; j < libraryLogs.Len(); j++ {
+			libraryLog := libraryLogs.At(j)
+			logs := libraryLog.LogRecords()
+			for k := 0; k < logs.Len(); k++ {
+				logMetadataMap := make(map[string]string)
+				resourceMapperMap := make(map[string]string)
+				log := logs.At(k)
+				// Copying resource attributes to log attributes
+				resourceLog.Resource().Attributes().Range(func(k string, v pcommon.Value) bool {
+					logAttributes := log.Attributes()
+					e.mergeAttributes(k, v, logAttributes)
+					return true
+				})
+				attributesMap := log.Attributes()
+				attributesMap.Range(func(key string, value pcommon.Value) bool {
+					if key == hostname {
+						resourceMapperMap[hostnameProperty] = value.Str()
+					}
+					logMetadataMap[key] = value.Str()
+					return true
+				})
+				e.logger.Debug("Sending log data", zap.String("body", log.Body().Str()), zap.Any("resourcemap", resourceMapperMap), zap.Any("metadatamap", logMetadataMap))
+				err := e.logIngestClient.SendLogs(ctx, log.Body().Str(), resourceMapperMap, logMetadataMap)
+				if err != nil {
+					e.logger.Error("error while exporting logs ", zap.Error(err), zap.String("body", log.Body().Str()), zap.Any("resourcemap", resourceMapperMap), zap.Any("metadatamap", logMetadataMap))
+				}
+				e.logger.Debug("Logs exported successfully!")
+			}
+		}
+	}
+	return nil
+}
+
+func (e *logExporter) mergeAttributes(k string, value pcommon.Value, logAttr pcommon.Map) {
+	switch value.Type() {
+	case pcommon.ValueTypeInt:
+		logAttr.PutInt(k, value.Int())
+	case pcommon.ValueTypeBool:
+		logAttr.PutBool(k, value.Bool())
+	case pcommon.ValueTypeDouble:
+		logAttr.PutDouble(k, value.Double())
+	case pcommon.ValueTypeStr:
+		logAttr.PutStr(k, value.Str())
+	case pcommon.ValueTypeMap:
+		values := map[string]interface{}{}
+		value.Map().Range(func(k string, v pcommon.Value) bool {
+			values[k] = v
+			return true
+		})
```

Comment:
Not sure it is intentional but it appears you've missed adding this back to `logAttr`, is that intentional?